mayastor/tests/bdev_test.rs

```diff
@@ -0,0 +1,337 @@
+        })
+        .await
+        .unwrap();
+
+    endpoint.into_inner().device_uri
+}
+
+/// create the work -- which means the nexus, replica's and the jobs. on return
+/// IO flows through mayastorTest to all 3 containers
+async fn create_topology(queue: Arc<JobQueue>) {
+    let r1 = create_target("ms1").await;
+    let r2 = create_target("ms2").await;
+    let endpoint = create_nexus("ms3", vec![r1, r2]).await;
+
+    // the nexus is running on ms3 we will use a 4th instance of mayastor to
+    // create a nvmf bdev and push IO to it.
+
+    let ms = MAYASTOR.get().unwrap();
+    let bdev = ms
+        .spawn(async move {
+            let bdev = bdev_create(&endpoint).await.unwrap();
+            bdev
+        })
+        .await;
+
+    // start the workload by running a job on each core, this simulates the way
+    // the targets use multiple cores
+    ms.spawn(async move {
+        for c in Cores::count() {
+            let bdev = Bdev::lookup_by_name(&bdev).unwrap();
+            let job = io_driver::Builder::new()
+                .core(c)
+                .bdev(bdev)
+                .qd(64)
+                .io_size(512)
+                .build()
+                .await;
+
+            queue.start(job);
+        }
+    })
+    .await;
+}
+
+async fn check_nexus<F: FnOnce(ListNexusReply)>(checker: F) {
+    let mut ms3 = DOCKER_COMPOSE
+        .get()
+        .unwrap()
+        .grpc_handle("ms3")
+        .await
+        .unwrap();
+    let list = ms3.mayastor.list_nexus(Null {}).await.unwrap().into_inner();
+    checker(list)
+}
+
+/// kill replica issues an unshare to the container which more or less amounts
+/// to the same thing as killing the container.
+async fn kill_replica(container: &str) {
+    let t = DOCKER_COMPOSE.get().unwrap();
+    let mut hdl = t.grpc_handle(container).await.unwrap();
+
+    hdl.bdev
+        .unshare(CreateReply {
+            name: "disk0".to_string(),
+        })
+        .await
+        .unwrap();
+}
+
+#[allow(dead_code)]
+async fn pause_replica(container: &str) {
+    let t = DOCKER_COMPOSE.get().unwrap();
+    t.pause(container).await.unwrap();
+}
+
+#[allow(dead_code)]
+async fn unpause_replica(container: &str) {
+    let t = DOCKER_COMPOSE.get().unwrap();
+    t.thaw(container).await.unwrap();
+}
+
+#[allow(dead_code)]
+async fn kill_local(container: &str) {
+    let t = DOCKER_COMPOSE.get().unwrap();
+    let mut hdl = t.grpc_handle(container).await.unwrap();
+    hdl.bdev
+        .destroy(BdevUri {
+            uri: "malloc:///disk0".into(),
+        })
+        .await
+        .unwrap();
+}
+
+async fn list_bdevs(container: &str) {
+    let mut h = DOCKER_COMPOSE
+        .get()
+        .unwrap()
+        .grpc_handle(container)
+        .await
+        .unwrap();
+    dbg!(h.bdev.list(Null {}).await.unwrap());
+}
+
+#[tokio::test]
+async fn nvmf_bdev_test() {
+    let queue = Arc::new(JobQueue::new());
+
+    Config::get_or_init(|| Config {
+        nvme_bdev_opts: NvmeBdevOpts {
```

Comment:
why are we using a non default config? so we won't spend as long in the test waiting for the timeout?

Reply:
Yes, as im working (and others) on these test it's easy to change things here rather than in opts.rs as it requires a recompile of mayastor. this requires just a recompile of the test itself.